File: tests/fixtures/open_redirect/python/safe_relative_redirect.py
# Safe: query arg routed through `ensure_relative_url` which enforces a
# leading `/` and rejects scheme-prefixed values (relative-only path).
from flask import request, redirect


def ensure_relative_url(raw):
    if not isinstance(raw, str) or not raw.startswith("/") or raw.startswith("//"):
        return "/"
    return raw


def handler():
    target = request.args.get("next")
    safe = ensure_relative_url(target)
    return redirect(safe)

File: tests/fixtures/open_redirect/typescript/safe_relative_redirect.ts
// Safe: req.query.next routed through `ensureRelativeUrl` which enforces
// a leading `/` and rejects scheme-prefixed values (relative-only path).
function ensureRelativeUrl(raw: string): string {
    if (!raw.startsWith('/') || raw.startsWith('//')) {
        return '/';
    }
    return raw;
}

export function handler(req: any, res: any): void {
    const target: string = req.query.next;
    const safe: string = ensureRelativeUrl(target);
    res.redirect(safe);
}
